File: src/client/java/sidly/wynnadhoc/features/TradeMarketOverlay.java
package sidly.wynnadhoc.features;

import sidly.wynnadhoc.config.ConfigManager;
import sidly.wynnadhoc.config.catagories.SimpleFeatureToggles;
import sidly.wynnadhoc.config.gui.GuiElement;
import sidly.wynnadhoc.config.gui.HudElementManager;
import sidly.wynnadhoc.config.gui.TextHudComponent;

public class TradeMarketOverlay {
    private static SimpleFeatureToggles config() {
        return ConfigManager.INSTANCE.config.toggles;
    }

    private static final GuiElement overlay = new GuiElement(
            config().TMOverlayMain,
            new TextHudComponent(
                    config().TMOverlayTitle,
                    TradeMarketCore.INSTANCE::showOverlay,
                    TradeMarketCore.INSTANCE::updateDisplay
            )
    );

    public static void registerHudElements() {
        HudElementManager.register(overlay);
    }
}

File: src/client/java/sidly/wynnadhoc/features/TradeMarketOverlay.kt
package sidly.wynnadhoc.features

import sidly.wynnadhoc.config.ConfigManager
import sidly.wynnadhoc.config.gui.HudElementManager
import sidly.wynnadhoc.config.gui.GuiElement
import sidly.wynnadhoc.config.gui.TextHudComponent
import sidly.wynnadhoc.event.PreInitEvent

object TradeMarketOverlay {
    private val config get() = ConfigManager.INSTANCE.config.toggles

    private var overlay: GuiElement = GuiElement(
        config.TMOverlayMain,
        TextHudComponent(
            config.TMOverlayTitle,
            TradeMarketOverlay::showOverlay,
            TradeMarketOverlay::updateDisplay
        )
    )

    fun registerHudElements(empty: PreInitEvent) {
        HudElementManager.register(overlay)
    }

    fun showOverlay(): Boolean {
        return true
    }

    fun updateDisplay(): String {
        return "Trade Market Highlighters"
    }
}